src/wallet/db.cpp

```diff
@@ -52,6 +53,24 @@ void CheckUniqueFileid(const CDBEnv& env, const std::string& filename, Db& db)
         }
     }
 }
+
+bool LockEnvDirectory(const fs::path& env_path)
```

Comment:
A lot in this function seems to be a duplication from `LockDataDirectory` in `init.cpp`. What do you think about abstracting them out to a utility function in `util.cpp` instead?

Reply:
Done